java/daikon/chicory/Instrument24.java

```diff
@@ -0,0 +1,1757 @@
+    }
+
+    // The offset of the first parameter.
+    int param_offset = mgen.isStatic() ? 0 : 1;
+
+    // Assumes addInstrumentationAtEntry has been called which sets nonceLocal.
+    // iload
+    // Push the nonce.
+    assert minfo.nonceLocal != null : "@AssumeAssertion(nullness): can't get here if null";
+    newCode.add(LoadInstruction.of(TypeKind.INT, minfo.nonceLocal.slot()));
+
+    // iconst
+    // Push the MethodInfo index.
+    newCode.add(loadIntegerConstant(minfo.method_info_index, mgen));
+
+    // iconst
+    // anewarray
+    // Create an array of objects with elements for each parameter.
+    newCode.add(loadIntegerConstant(paramTypes.length, mgen));
+    ClassDesc objectArrayCD = objectCD.arrayType(1);
+    newCode.add(NewReferenceArrayInstruction.of(mgen.getPoolBuilder().classEntry(objectCD)));
+
+    // Put each parameter into the array.
+    int param_index = param_offset;
+    for (int ii = 0; ii < paramTypes.length; ii++) {
+      newCode.add(StackInstruction.of(Opcode.DUP));
+      newCode.add(loadIntegerConstant(ii, mgen));
+      ClassDesc at = paramTypes[ii];
+      if (at.isPrimitive()) {
+        createPrimitiveWrapper(newCode, at, param_index, mgen);
+      } else { // it's a reference of some sort
+        newCode.add(LoadInstruction.of(TypeKind.REFERENCE, param_index));
+      }
+      newCode.add(ArrayStoreInstruction.of(Opcode.AASTORE));
+      param_index += TypeKind.from(at).slotSize();
+    }
+
+    // If this is an exit, push the return value and line number.
+    // The return value is stored in the local "return__$trace2_val".
+    // If the return value is a primitive, wrap it in the appropriate wrapper.
+    if (methodToCall.equals("exit")) {
+      ClassDesc ret_type = mgen.getReturnType();
+      if (ret_type.equals(CD_void)) {
+        newCode.add(ConstantInstruction.ofIntrinsic(Opcode.ACONST_NULL));
+      } else {
+        LocalVariable return_local = getReturnLocal(mgen, ret_type, minfo);
+        if (ret_type.isPrimitive()) {
+          createPrimitiveWrapper(newCode, ret_type, return_local.slot(), mgen);
+        } else {
+          newCode.add(LoadInstruction.of(TypeKind.REFERENCE, return_local.slot()));
+        }
+      }
+
+      // push line number
+      newCode.add(loadIntegerConstant(line, mgen));
+    }
+
+    MethodTypeDesc methodArgs;
+    // Call the specified method.
+    if (methodToCall.equals("exit")) {
+      methodArgs =
+          MethodTypeDesc.of(CD_void, CD_Object, CD_int, CD_int, objectArrayCD, CD_Object, CD_int);
+    } else {
+      methodArgs = MethodTypeDesc.of(CD_void, CD_Object, CD_int, CD_int, objectArrayCD);
+    }
+    MethodRefEntry mre = mgen.getPoolBuilder().methodRefEntry(runtimeCD, methodToCall, methodArgs);
+    newCode.add(InvokeInstruction.of(Opcode.INVOKESTATIC, mre));
+  }
+
+  /** Possibly modified default switch target. */
+  private Label modifiedTarget;
+
+  /** Possibly modified switch case list. */
+  private List<SwitchCase> modifiedCaseList;
```

Comment:
These two fields can be local variables.  `retargetStartLabel(Label defaultTarget, List<SwitchCase> caseList, MInfo24 minfo)` can return (instead of a boolean) either null or a struct containing these two fields.